contrib/database/sql/sql.go

```diff
@@ -236,3 +232,29 @@ func Open(driverName, dataSourceName string, opts ...Option) (*sql.DB, error) {
 	}
 	return OpenDB(&dsnConnector{dsn: dataSourceName, driver: d}, opts...), nil
 }
+
+func processOptions(cfg *config, driverName string, opts ...Option) {
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	checkDBMPropagation(cfg, driverName)
+}
+
+func checkDBMPropagation(cfg *config, driverName string) {
+	fullModeSupported := func() bool {
```

Comment:
Just curious about style: why make this an inner anonymous function rather than a named function?

Reply:
it was just a matter of namespacing (this way I could use a shorter function name instead of `dbmPropagationFullModeSupported`). Anyway, I refactored this code, please take a look at the new version 🙏